### ISSUE TITLE
EHOSTUNREACH crash

### ISSUE THREAD

**Issue by achingbrain** [libp2p/js-libp2p#883]
- **Version**: 0.30.x
- **Platform**: os x
- **Subsystem**: ??

#### Steps to reproduce the error:

1. start a node
2. wait

```
Error: connect EHOSTUNREACH 2604:1380:1000:b500::1:4002 - Local (:::60783)
    at internalConnect (net.js:921:16)
    at defaultTriggerAsyncIdScope (internal/async_hooks.js:430:12)
    at net.js:1009:9
    at processTicksAndRejections (internal/process/task_queues.js:75:11) {
  errno: -65,
  code: 'EHOSTUNREACH',
  syscall: 'connect',
  address: '2604:1380:1000:b500::1',
  port: 4002
}
```

May be related to #844 - something, somewhere is not handling errors properly.

**Comment by vasco-santos** [comment]
Yes, I could replicate that error in https://github.com/libp2p/js-libp2p/issues/844#issuecomment-760230337

A temporary solution is https://github.com/libp2p/js-libp2p-websockets/pull/118

This should be fixed in node soon as it was already merged a fix, but we should still merge the websockets fix while it is not released and people update node:

- https://github.com/websockets/ws/issues/1835
- https://github.com/nodejs/node/issues/36931#event-4287780957

**Comment by vasco-santos** [comment]
I released `libp2p-websockets@0.15.2` with the error handler for this.

I am going to close the issue, feel free to reopen if this is still an issue with the update.
